src/node_url.cc

```diff
@@ -829,6 +829,16 @@ static url_host_type ParseOpaqueHost(url_host* host,
   return type;
 }
 
+static inline bool IsAllASCII(std::string* input) {
+  for (size_t n = 0; n < input->size(); n++) {
+    const char ch = (*input)[n];
+    if (ch & 0x80) {
```

Comment:
Please declare a new CHAR_TEST in the dedicated section above (grep CHAR_TEST to find where it is):

```c++
// https://infra.spec.whatwg.org/#ascii-code-point
CHAR_TEST(8, IsASCIICodePoint, (ch >= '\0' && ch <= '\x7f'))
```

And use `IsASCIICodePoint(ch)` here.